### DISCUSSION TITLE
Feature Request: Auto Update

### DISCUSSION THREAD

**Discussion by MisterSilvereagle** [valinet/ExplorerPatcher#3205] [Ideas]
I think the title says it all. 

I would love an option to enable ExplorerPatcher automatically downloading and installing new updates as soon as it finds them during Windows Startup. 

I slowly get bored by right-clicking the taskbar, selecting "Properties", selecting "Updates" and clicking "Update and restart" everytime I get a push notification about an available update...

**Comment by Amrsatrio** [comment]
Or what about a button on the notification so that one click there is what it takes to update EP...

Personally I think auto updates can introduce a security issue.

**Comment by pyrates999** [comment]
I think in order for that to happen, EP needs to have its ep_setup.exe file signed with a valid code signing certificate.  Otherwise even if auto update is enabled, windows will still prompt you because it's not signed.

**Comment by Amrsatrio** [comment]
What about this?

![image](https://github.com/valinet/ExplorerPatcher/assets/20662640/28d9fd28-3b01-4b47-92ba-583019cff349)

**Reply by MisterSilvereagle** [reply]
That would be really great!

Currently, it doesn't have this button...
![EP-Update](https://github.com/valinet/ExplorerPatcher/assets/58659498/57684206-3656-46ed-bf99-29bb980d0468)

**Reply by cheeseandjamsandwich** [reply]
I also would really like an Auto-Update option...
Especially when i'm setting up ep on my parent's computer (Gotta have the Quick Launch toolbar for them!)...  At the moment, getting my dad to find his way to EP's properties dialog and the Updates tab to do the update is stretching it a bit.

The above idea of having a big Update Now button is very decent compromise.

It's far more likely for my parents to face a problem of EP being old, not updated, clashing with Window's updates, than with EP having troubles it can't handle itself with the updates.

Auto-update option... OR  Update Now button when a new version is found... Either would be very much appreciated!

**Reply by pyrates999** [reply]
I think auto update that is done automatically would require EP to be signed with a valid certificate first.
